src/main/java/de/uhd/ifi/se/decision/management/jira/quality/MetricCalculator.java

```diff
@@ -0,0 +1,402 @@
+	linkissues.add(jiraIssue);
+	linkDistanceMap.put(0, linkissues);
+	List<Issue> inMap = new ArrayList<Issue>();
+	inMap.add(jiraIssue);
+	for (int i = 1; i <= linkDistance; i++) {
+	    linkissues = new ArrayList<Issue>();
+	    for (Issue issue : linkDistanceMap.get(i - 1)) {
+		Collection<Issue> issueColl = issueLinkManager.getLinkCollection(issue, user).getAllIssues();
+		linkissues.addAll(issueColl);
+		linkissues.removeAll(inMap);
+	    }
+	    List<Issue> linkissueswithoutduplicate = new ArrayList<>(new HashSet<>(linkissues)); // Remove Duplicates
+	    linkDistanceMap.put(i, linkissueswithoutduplicate);
+	    inMap.addAll(linkissueswithoutduplicate);
+	}
+	return linkDistanceMap;
+    }
+
+    private List<Issue> getJiraIssuesForProject(long projectId) {
+	List<Issue> jiraIssues = new ArrayList<Issue>();
+	JqlClauseBuilder jqlClauseBuilder = JqlQueryBuilder.newClauseBuilder();
+	Query query = jqlClauseBuilder.project(projectId).buildQuery();
+	SearchResults<Issue> searchResults = null;
+	SearchService searchService = ComponentAccessor.getComponentOfType(SearchService.class);
+	try {
+	    searchResults = searchService.search(user, query, PagerFilter.getUnlimitedFilter());
+	    jiraIssues = JiraSearchServiceHelper.getJiraIssues(searchResults);
+	} catch (SearchException e) {
+	    LOGGER.error("Getting JIRA issues for project failed. Message: " + e.getMessage());
+	}
+	return jiraIssues;
+    }
+
+    private Map<String, List<DecisionKnowledgeElement>> getDecisionKnowledgeElementsFromCode(String projectKey) {
+	// Extracts Decision Knowledge from Code Comments AND Commits
+	GitDecXtract gitExtract = new GitDecXtract(projectKey);
+	Map<String, List<DecisionKnowledgeElement>> elementsMap = gitExtract.getElementsSplitByCodeAndCommit("develop");
+	return elementsMap;
+    }
+
+    public Map<String, Integer> numberOfCommentsPerIssue() {
+	Map<String, Integer> numberMap = new HashMap<String, Integer>();
+	int numberOfComments;
+	for (Issue jiraIssue : jiraIssues) {
+	    try {
+		numberOfComments = ComponentAccessor.getCommentManager().getComments(jiraIssue).size();
+	    } catch (NullPointerException e) {
+		LOGGER.error("Getting number of comments for JIRA issues failed. Message: " + e.getMessage());
+		numberOfComments = 0;
+	    }
+	    numberMap.put(jiraIssue.getKey(), numberOfComments);
+	}
+	return numberMap;
+    }
+
+    public Map<String, Integer> numberOfCommitsPerIssue() {
+	Map<String, Integer> resultMap = new HashMap<String, Integer>();
+	try {
+	    GitClient gitClient = new GitClientImpl(projectKey);
+	    if (gitClient != null) {
+		for (Issue jiraIssue : jiraIssues) {
+		    int numberOfCommits = gitClient.getNumberOfCommits(jiraIssue);
+		    resultMap.put(jiraIssue.getKey(), numberOfCommits);
+		}
+	    }
+	} catch (Exception ex) {
+	    LOGGER.error(ex.getMessage());
+	}
+	return resultMap;
+    }
+
+    public Map<String, Integer> getNumberOfDecisionKnowledgeElementsForJiraIssues(KnowledgeType type,

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getNumberOfDecisionKnowledgeElementsForJiraIssues(KnowledgeType, Integer)' has an NPath complexity of 246](https://app.codacy.com/app/UHD/cures-condec-jira/pullRequest?prid=4666670)